### PR TITLE
HITL - Add unit testing.

### PR DESCRIPTION
**Pull request by 0mdc**
## Motivation and Context

This changeset adds unit testing for `habitat-hitl`.

The following tests are added:
* `main()` smoke test.
* Example apps smoke tests.
* User mask test.

## How Has This Been Tested

Tested locally and on CI.

## Types of changes

- **\[Development\]**

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

**Comment by 0mdc** [comment]
This looks stable. However, `test/test_humanoid.py` is failing. I'm checking whether this is somehow related to my changes.

File: .circleci/config.yml
Comment (0mdc) [inline .circleci/config.yml:286]: I'll change this before merging. Waiting for CI to pass.

File: test/hitl/test_hitl_examples.py
Comment (0mdc) [inline test/hitl/test_hitl_examples.py:38]: This uses the [Hydra experiment workflow](https://hydra.cc/docs/patterns/configuring_experiments/). I'm intending to have more than one in the future.

**Review by 0mdc — COMMENTED** [review]
Comments for reviewers.

File: .circleci/config.yml
Comment (aclegg3) [inline .circleci/config.yml:274]: Any story here? Was calling pytest from module failing?

File: .circleci/config.yml
Comment (aclegg3) [inline .circleci/config.yml:282]: Isn't this link already created in api test job? Are you expecting concurrency?

**Review by aclegg3 — APPROVED** [review]
Thanks for adding tests. Small questions on the config changes, but not a blocker if things are working as expected.

File: .circleci/config.yml
Comment (0mdc) [reply .circleci/config.yml:274]: This was not failing - just wanted to standardize test calls. I'll use the other form to avoid unnecessary changes.

File: .circleci/config.yml
Comment (0mdc) [reply .circleci/config.yml:282]: You're correct. It should work without the previous step.
I'm not expecting concurrency.

**Pull request merged**
